cli/azd/pkg/output/ux/environment_details.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package ux
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/fatih/color"
+)
+
+type EnvironmentDetails struct {
+	Subscription string
+	Location     string
+}
+
+func (t *EnvironmentDetails) ToString(currentIndentation string) string {
+	return fmt.Sprintf(
+		"Subscription: %s\nLocation: %s\n",
+		color.BlueString(t.Subscription),
+		color.BlueString(t.Location),
+	)
+}
+
+func (t *EnvironmentDetails) MarshalJSON() ([]byte, error) {
+	// reusing the same envelope from console messages
+	return json.Marshal(output.EventForMessage(fmt.Sprintf("\n%s\n%s\n", t.Subscription, t.Location)))
+}
```

Comment:
This struct feels a bit overkill if only used to print out the sub/location.

Reply:
This makes it easier to re-use the same output if we ever want it as part of other command..  (like azd up) or something else.
UX components allows re-use and easy way to test and update output away from the component that uses it